pandas/core/ops.py

```diff
@@ -1435,6 +1436,82 @@ def add_flex_arithmetic_methods(cls):
     add_methods(cls, new_methods=new_methods)
 
 
+# ----------------------------------------------------------------------
+# Unary
+
+def _unary_method(cls, op, special=True):
+    op_name = _get_op_name(op, special)
+
+    def f(self):
+        if self.ndim > 2:
+            with np.errstate(all='ignore'):
+                # apply op at once based on current impl
+                new_data = op(self._values)
```

Comment:
should this be done block/column-wise?  in particular casting to object often messes up on datetime64 columns